File: src/test/java/org/geepawhill/contentment/test/JavaFxTest.java
package org.geepawhill.contentment.test;

import javafx.geometry.VPos;
import javafx.scene.Group;
import javafx.scene.text.Text;
import javafx.scene.text.TextAlignment;
import org.geepawhill.contentment.geometry.Point;
import org.geepawhill.contentment.geometry.PointPair;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.geepawhill.contentment.test.ContentmentAssertions.assertThat;

public class JavaFxTest extends ContentmentTest
{

	@BeforeEach
	public void before()
	{
		
	}
	
	@Test
	public void whenIsBoundsActive()
	{
		Group group = new Group();
		Text text = new Text("Here's some text.");
		text.setTextAlignment(TextAlignment.LEFT);
		text.setTextOrigin(VPos.TOP);
		text.setTranslateX(100);
		text.setTranslateY(0);
		System.out.println(new PointPair(text));
		assertThat(new PointPair(text).getFrom()).isEqualTo(new Point(100,0));
		group.getChildren().add(text);
		
	}
}

File: src/test/kotlin/org/geepawhill/contentment/test/JavaFxTest.kt
package org.geepawhill.contentment.test

import javafx.geometry.VPos
import javafx.scene.Group
import javafx.scene.text.Text
import javafx.scene.text.TextAlignment
import org.geepawhill.contentment.geometry.Point
import org.geepawhill.contentment.geometry.PointPair
import test.ContentmentAssertions.Companion.assertThat
import org.junit.jupiter.api.Test

class JavaFxTest : ContentmentTest() {

    @Test
    fun whenIsBoundsActive() {
        val group = Group()
        val text = Text("Here's some text.")
        text.textAlignment = TextAlignment.LEFT
        text.textOrigin = VPos.TOP
        text.translateX = 100.0
        text.translateY = 0.0
        println(PointPair(text))
        assertThat(PointPair(text).from).isEqualTo(Point(100.0, 0.0))
        group.children.add(text)

    }
}
